database/pgsql/layer.go

```diff
@@ -238,28 +250,51 @@ func (pgSQL *pgSQL) InsertLayer(layer database.Layer) error {
 
 	// Get parent ID.
 	var parentID zero.Int
+	mapNamespaceIDs := make(map[string]int)
 	if layer.Parent != nil {
 		if layer.Parent.ID == 0 {
 			log.Warning("Parent is expected to be retrieved from database when inserting a layer.")
 			return cerrors.NewBadRequestError("Parent is expected to be retrieved from database when inserting a layer.")
 		}
 
 		parentID = zero.IntFrom(int64(layer.Parent.ID))
-	}
 
-	// Find or insert namespace if provided.
-	var namespaceID zero.Int
-	if layer.Namespace != nil {
-		n, err := pgSQL.insertNamespace(*layer.Namespace)
+		// Find its parent's namespaces
+		rows, err := pgSQL.Query(searchLayerNamespace, parentID)
```

Comment:
Why layer.Parent.Namespaces aren't already loaded? We already make the assumption that the provided Parent layer has been retrieved from the database (see L255).